tornettools/generate.py

```diff
@@ -133,14 +141,57 @@ def __get_scaled_tgen_server_bandwidth_kbit(args):
     scaled_bw = scaled_client_bw * n_clients_per_server
     return scaled_bw
 
-def __server(args, server):
+def __filter_nodes(network, ip_address_hint, country_code_hint):
+    all_nodes = [{'id': node[0], **node[1]} for node in network.nodes(data=True)]
+
+    if ip_address_hint is not None and not ip_address_hint.is_global:
+        # ignore the hint if the IP address is not global
```

Comment:
Does "global" mean it's a globally routable address, compared to a local address such as those in the `192.168.0.0/16` range?

Reply:
Local addresses should be considered "not global". The address is considered global if it's not in any of the following networks:

```
_public_network = IPv4Network('100.64.0.0/10')
_private_networks = [
    IPv4Network('0.0.0.0/8'),
    IPv4Network('10.0.0.0/8'),
    IPv4Network('127.0.0.0/8'),
    IPv4Network('169.254.0.0/16'),
    IPv4Network('172.16.0.0/12'),
    IPv4Network('192.0.0.0/29'),
    IPv4Network('192.0.0.170/31'),
    IPv4Network('192.0.2.0/24'),
    IPv4Network('192.168.0.0/16'),
    IPv4Network('198.18.0.0/15'),
    IPv4Network('198.51.100.0/24'),
    IPv4Network('203.0.113.0/24'),
    IPv4Network('240.0.0.0/4'),
    IPv4Network('255.255.255.255/32'),
]
```